synapse/api/filtering.py

```diff
@@ -368,13 +370,16 @@ def check(self, event):
 
             room_id = event.get("room_id", None)
             ev_type = event.get("type", None)
-            is_url = "url" in event.get("content", {})
+
+            content = event.get("content", {})
+            # check if there is a string url field in the content for filtering purposes
+            contains_url = isinstance(content.get("url"), text_type)
```

Comment:
the spec says "includes only events with a url key in their content"

it doesn't say it has to be a string...

Reply:
(this might be a good opportunity to clarify the spec so it makes sense)

Reply:
well, if there are compelling reasons to do so, then sure, let's change the spec. Are there, though?

[if you actually want to check it's a string, you probably want `six.string_types` rather than `six.text_type`]

Reply:
There's not really a use case for a singular-named variable to contain an object/array, and it's somewhat implied it is a string anyways. Might as well just say it is a string.

Reply:
ok, but (a) that's not quite the same as 'filters have to check for the stringiness' and (b) "might as well" doesn't sound worth the effort of changing the spec to me.

Reply:
@richvdh, existing code disagrees: https://github.com/matrix-org/synapse/blob/bc80b3f454aa9b9ca8bc710ff502b83892ac0a91/synapse/storage/events.py#L1575-L1577